### ISSUE TITLE
series.toArray() method throws `Error: assertion `left == right` failed: impl error`

### ISSUE THREAD

**Issue by Flyinchicken** [pola-rs/nodejs-polars#242]
#### Have you tried latest version of polars?

- yes

#### What version of polars are you using?

0.13.0

#### What operating system are you using polars on?

Microsoft Windows 11 (wsl Ubuntu version 2)

#### What node version are you using
node 20.13.1 

#### Describe your bug.

series.toArray() method throws `Error: assertion 'left == right' failed: impl error`

#### What are the steps to reproduce the behavior?

```ts
async createFromSeries(colSeries: Series) {
          colSeries.toArray()
...
}
```

I am trying to convert a series to an array for storing it in MongoDB. This method did not work for some of the columns so the above code might not be sufficient to reproduce the bug.

It might be related to this [issue](https://github.com/pola-rs/polars/issues/12188)

#### What is the actual behavior?

```
thread '<unnamed>' panicked at /root/.cargo/git/checkouts/polars-b0d90607192fd414/7bc7014/crates/polars-core/src/series/iterator.rs:74:9:
@databank/api:dev: assertion `left == right` failed: impl error
@databank/api:dev:   left: 10
@databank/api:dev:  right: 1
@databank/api:dev: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
@databank/api:dev: [Nest] 73664  - 07/10/2024, 3:30:10 PM   ERROR [ExceptionsHandler] assertion `left == right` failed: impl error
@databank/api:dev:   left: 10
@databank/api:dev:  right: 1
@databank/api:dev: Error: assertion `left == right` failed: impl error
@databank/api:dev:   left: 10
@databank/api:dev:  right: 1
@databank/api:dev:     at Proxy.toArray
```

**Comment by Bidek56** [comment]
We have many tests what use `toArray()` and they all work fine, can you please provide a MRE? Thx

**Comment by Flyinchicken** [comment]
Hi, I have created a test dataset with 1 column and 23375 entries. A column called "pin" with floats from 1.0 to 862.0 with repetition.

Below is the code in TypeScript I used to recreate the issue:
```ts
df = pl.readCSV(csvString, { tryParseDates: true });

console.log(csvString);
console.log(df.columns);
console.log(df['pin'].isFloat());
console.log(df['pin'].name)
console.log(df['pin'].toArray())
```
Here is the error message:
```
@databank/api:dev: 859.0
@databank/api:dev: 859.0
@databank/api:dev: 860.0
@databank/api:dev: 860.0
@databank/api:dev: 861.0
@databank/api:dev: 861.0
@databank/api:dev: 862.0
@databank/api:dev: 862.0
@databank/api:dev: 
@databank/api:dev: [ 'pin' ]
@databank/api:dev: true
@databank/api:dev: pin
@databank/api:dev: thread '<unnamed>' panicked at /root/.cargo/git/checkouts/polars-b0d90607192fd414/7bc7014/crates/polars-core/src/series/iterator.rs:74:9:
@databank/api:dev: assertion `left == right` failed: impl error
@databank/api:dev:   left: 12
@databank/api:dev:  right: 1
@databank/api:dev: note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
@databank/api:dev: [Nest] 29298  - 07/11/2024, 11:48:16 AM   ERROR [ExceptionsHandler] assertion `left == right` failed: impl error
@databank/api:dev:   left: 12
@databank/api:dev:  right: 1
@databank/api:dev: Error: assertion `left == right` failed: impl error
@databank/api:dev:   left: 12
@databank/api:dev:  right: 1
@databank/api:dev:     at Proxy.toArray (~/node_modules/.pnpm/nodejs-polars@0.13.0/node_modules/nodejs-polars/bin/series/index.js:605:23)
```

I have tried to read the same file in Python and the `to_list()` method works without any issue:
```python
import polars as pl

df = pl.read_csv(
    r"./data.csv",
      try_parse_dates=True
)

print(df['pin'].to_list())
```
Output:
```
...
.0, 832.0, 832.0, 832.0, 832.0, 832.0, 832.0, 834.0, 834.0, 834.0, 834.0, 834.0, 834.0, 834.0, 835.0, 835.0, 835.0, 835.0, 835.0, 835.0, 835.0, 835.0, 836.0, 836.0, 836.0, 836.0, 836.0, 837.0, 837.0, 837.0, 837.0, 837.0, 837.0, 837.0, 837.0, 838.0, 838.0, 838.0, 838.0, 838.0, 838.0, 838.0, 838.0, 838.0, 838.0, 838.0, 839.0, 839.0, 839.0, 839.0, 839.0, 839.0, 839.0, 839.0, 840.0, 840.0, 840.0, 840.0, 840.0, 842.0, 842.0, 842.0, 842.0, 842.0, 842.0, 843.0, 843.0, 843.0, 843.0, 843.0, 843.0, 843.0, 843.0, 843.0, 844.0, 844.0, 844.0, 844.0, 845.0, 845.0, 845.0, 845.0, 845.0, 845.0, 845.0, 846.0, 846.0, 846.0, 847.0, 847.0, 847.0, 847.0, 847.0, 848.0, 848.0, 848.0, 848.0, 848.0, 848.0, 849.0, 849.0, 849.0, 849.0, 849.0, 850.0, 850.0, 850.0, 851.0, 851.0, 851.0, 851.0, 851.0, 851.0, 851.0, 851.0, 851.0, 852.0, 852.0, 852.0, 852.0, 852.0, 852.0, 852.0, 853.0, 853.0, 853.0, 854.0, 854.0, 855.0, 855.0, 856.0, 856.0, 856.0, 857.0, 857.0, 857.0, 857.0, 857.0, 857.0, 858.0, 858.0, 859.0, 859.0, 859.0, 860.0, 860.0, 861.0, 861.0, 862.0, 862.0]
```
Thank you!

**Comment by universalmind303** [comment]
@Flyinchicken can you provide a minimum reproducible example (MRE). Either by sharing the CSV, or sharing code to create the same CSV or dataframe. 

Without one, it's hard/impossible to debug the issue. Thanks!

**Comment by Flyinchicken** [comment]
[data.csv](https://github.com/user-attachments/files/16181229/data.csv)
Please see the attached csv that I used in the above example. Thank you!

**Comment by universalmind303** [comment]
@Flyinchicken I just released `0.14.0`, which includes a fix for this
